Helper/Cart.php

```diff
@@ -2266,7 +2267,29 @@ public function hasProductRestrictions($quote = null)
      */
     public function createCartByRequest($request)
     {
-        return $this->createCart($request['items'], $request['metadata']);
+        $options = json_decode($request['items'][0]['options'], true);
+        $storeId = $options['storeId'];
+
+        // try returning from cache
+        if ($isBoltOrderCachingEnabled = $this->isBoltOrderCachingEnabled($storeId)) {
+            $cacheIdentifier = $this->getCartCacheIdentifier($request);
```

Comment:
Could you please add a comment on why we know that request is unique for each user (because it contains Magento form id).
It can be a hard moment for code readers.

Reply:
done